ui/src/source.ts

```diff
@@ -112,15 +117,24 @@ export enum RevisionType {
   Sha = "sha",
 }
 
+// Client representation of a git branch
 export interface Branch {
   type: RevisionType.Branch;
   name: string;
-  peerId?: string;
+  // The id of the project this branch belongs to
+  projectId: string;
+  // The id of the peer this branch belongs to
+  peerId: string;
 }
 
+// Client representation of a git tag
 export interface Tag {
   type: RevisionType.Tag;
   name: string;
+  // The id of the project this tag belongs to
+  peerId: string;
+  // The id of the project this tag belongs to
+  projectId: string;
```

Comment:
I think they should stay separate for now because the `currentRevision` can change independently of the `currentPeerId` - e.g. if the user selects a revision that belongs to the same peer.